test/src/unit-capi-fragment_info.cc

```diff
@@ -351,6 +351,17 @@ TEST_CASE(
   rc = tiledb_fragment_info_get_fragment_name(ctx, fragment_info, 1, &name);
   CHECK(rc == TILEDB_OK);
 
+  tiledb_string_t* nameStr;
+  rc = tiledb_fragment_info_get_fragment_name_v2(
+      ctx, fragment_info, 1, &nameStr);
+  CHECK(rc == TILEDB_OK);
+  const char* nameStrData;
+  size_t nameStrLength;
+  rc = tiledb_string_view(nameStr, &nameStrData, &nameStrLength);
+  CHECK(rc == TILEDB_OK);
+  CHECK(std::string(name) == std::string(nameStrData));
```

Comment:
What is this check doing? Is it checking the old version of the function? Or the new version of the function?

In particular, what happens when the old function is removed? Does this test break?
If this test should persist past the removal of the original function, it should be rewritten to not use the deprecated function. If it doesn't need to, then it should be isolated or removed along with any test on the deprecated version of the function.